src/ast/keys.ts

```diff
@@ -1,8 +1,15 @@
-const keys: {
+import { GenericEsTreeNode } from './nodes/shared/Node';
+
+export const keys: {
 	[name: string]: string[];
 } = {
 	Program: ['body'],
 	Literal: []
 };
 
-export default keys;
+export function getAndCreateKeys(esTreeNode: GenericEsTreeNode) {
```

Comment:
This would be more elegant but, as `Node` construction is one of the most performance critical tasks, I wanted to prevent the function call if possible. The checking logic is still only performed once, albeit in `Node.parseNode`.